src/wepy/hdf5.py

```diff
@@ -2295,6 +2299,20 @@ def _add_run_field(self, run_idx, field_path, data, sparse_idxs=None,
 
         """
 
+        # TODO, SNIPPET: check that we have the right permissions
```

Comment:
The concatenate feature wasn't used much and was enforced rather sporadically. I moved that checking code from the `compute_observable` here as a snippet so it isn't lost.